bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/internal/BroadlinkHandlerFactory.java

```diff
@@ -0,0 +1,115 @@
+        return BroadlinkBindingConstants.SUPPORTED_THING_TYPES_UIDS_TO_NAME_MAP.keySet().contains(thingTypeUID);
+    }
+
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+        if (logger.isDebugEnabled()) {
+            logger.debug("Creating Thing handler for '{}'", thingTypeUID.getAsString());
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_RM)) {
+            return new BroadlinkRemoteModel2Handler(thing, commandDescriptionProvider);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_RM2)) {
+            logger.debug("RM 2 handler requested created");
+            return new BroadlinkRemoteModel2Handler(thing, commandDescriptionProvider);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_RM3)) {
+            logger.debug("RM 3 handler requested created");
+            return new BroadlinkRemoteHandler(thing, commandDescriptionProvider);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_RM3Q)) {
+            logger.debug("RM 3 v11057 handler requested created");
+            return new BroadlinkRemoteModel3V44057Handler(thing, commandDescriptionProvider);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_RM4)) {
+            logger.debug("RM 4 handler requested created");
+            return new BroadlinkRemoteModel4Handler(thing, commandDescriptionProvider);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_A1)) {
+            logger.debug("A1 handler requested created");
+            return new BroadlinkA1Handler(thing);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_MP1)) {
+            return new BroadlinkStripModel1Handler(thing);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_MP1_1K3S2U)) {
+            return new BroadlinkStripModel11K3S2UHandler(thing);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_SP1)) {
+            return new BroadlinkSocketModel1Handler(thing);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_SP2)) {
+            return new BroadlinkSocketModel2Handler(thing);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_SP3)) {
+            return new BroadlinkSocketModel3Handler(thing);
+        }
+        if (thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_MP2)) {
+            return new BroadlinkStripModel1Handler(thing);
+        } else {
+            // thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_S1C);
+            // thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_PIR);
+            // thingTypeUID.equals(BroadlinkBindingConstants.THING_TYPE_MAGNET);
+            logger.error("Can't create handler for thing type UID: {}", thingTypeUID.getAsString());
```

Comment:
warn level? please review all cases of logger.error use the find and replace on it and it should be quick enough compared to me commenting on it over and over.